packages/syft/src/syft/core/smpc/protocol/beaver/beaver.py

```diff
@@ -0,0 +1,289 @@
+"""Beaver Triples Protocol.
+
+D. Beaver. *Efficient multiparty protocols using circuit randomization*.
+In J. Feigenbaum, editor, CRYPTO, volume **576** of Lecture Notes in
+Computer Science, pages 420–432. Springer, 1991.
+"""
+
+
+# stdlib
+import operator
+import secrets
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+# third party
+import numpy as np
+
+# relative
+from ....tensor.smpc.mpc_tensor import MPCTensor
+from ....tensor.smpc.share_tensor import ShareTensor
+from ...store import register_primitive_generator
+from ...store import register_primitive_store_add
+from ...store import register_primitive_store_get
+from ...store.exceptions import EmptyPrimitiveStore
+
+ttp_generator = np.random.default_rng()
+
+
+def _get_triples(
+    op_str: str,
+    nr_parties: int,
+    a_shape: Tuple[int],
+    b_shape: Tuple[int],
+    ring_size: int = 2 ** 32,
+    **kwargs: Dict[Any, Any],
+) -> Tuple[Tuple[Tuple[ShareTensor, ShareTensor, ShareTensor]]]:
+    """Get triples.
+
+    The Trusted Third Party (TTP) or Crypto Provider should provide this triples Currently,
```

Comment:
<3 the docs

Reply:
@gmuraru  did the docs in SyMPC, It was really helpful for me also.